Be clearer about message error vs. stream errors (#205)
ietf-wg-masque/draft-ietf-masque-h3-datagram: draft-ietf-masque-h3-datagram.md

## Error Handling

Unless otherwise stated, Capsule Protocol endpoints are expected to apply the
same message format rules as the underlying HTTP version. When a receiver
encounters a processing error, it MUST be treated as if it were a malformed or
incomplete message. For HTTP/3, the handling of malformed messages is described
in {{Section 4.1.3 of H3}}. For HTTP/2, the handling of malformed messages is
described in {{Section 8.1.1 of H2}}. For HTTP/1.1, the handling of incomplete
messages is described in {{Section 8 of H1}}.

> For HTTP/2 and HTTP/3, once a both endpoints agree to use the Capsule Protocol,

Reviewer: ```suggestion
For HTTP/2 and HTTP/3, once both endpoints agree to use the Capsule Protocol,
```

the frame usage requirements of the stream change. In HTTP/2, only DATA or
stream management frames (RST_STREAM, WINDOW_UPDATE, and PRIORITY) are
permitted; other frame types MUST NOT be sent and MUST be treated as a stream
error ({{Section 5.4.2 of H2}}) if received. In HTTP/3, only DATA frames are
permitted; other frame types MUST NOT be sent and MUST be treated as connection

> error o)f type H3_FRAME_UNEXPECTED ({{Section 8 of H3}}) if received. In both

Reviewer 2: ```suggestion
error of type H3_FRAME_UNEXPECTED ({{Section 8 of H3}}) if received. In both
```

## Error Handling

> Unless otherwise stated, Capsule Protocol endpoints are expected to apply the
> same message format rules as the underlying HTTP version. When a receiver
> encounters a processing error, it MUST be treated as if it were a malformed or
> incomplete message. For HTTP/3, the handling of malformed messages is described

Reviewer 3: I'm not sure this first sentence adds much value, because it sounds to me like "please follow referenced specs unless told not to"

Also I'd specifically call out the capsule protocol in the normative sentence that follows

```suggestion
When a receiver encounters an error processing the Capsule Protocol, the receiver
MUST treat it as if it had received a malformed or incomplete HTTP message.
For HTTP/3, the handling of malformed messages is described
```

Author: With the change below, this suggestion is better, thanks.

in {{Section 4.1.3 of H3}}. For HTTP/2, the handling of malformed messages is
described in {{Section 8.1.1 of H2}}. For HTTP/1.1, the handling of incomplete
messages is described in {{Section 8 of H1}}.

> For HTTP/2 and HTTP/3, once both endpoints agree to use the Capsule Protocol,
> the frame usage requirements of the stream change. In HTTP/2, only DATA or
> stream management frames (RST_STREAM, WINDOW_UPDATE, and PRIORITY) are
> permitted; other frame types MUST NOT be sent and MUST be treated as a stream
> error ({{Section 5.4.2 of H2}}) if received. In HTTP/3, only DATA frames are
> permitted; other frame types MUST NOT be sent and MUST be treated as connection
> error of type H3_FRAME_UNEXPECTED ({{Section 8 of H3}}) if received. In both
> cases, extension frame types MAY be used if specifically permitted by the
> definition of the extension.

Reviewer 3: This text duplicates normative text from our references, how about the below? Also, I'd move it to the end of `{{capsule-protocol}}` since it isn't specific to error handling.

```suggestion
Since the Capsule Protocol only applies to definitions of new HTTP Upgrade
Tokens, in HTTP/2 and HTTP/3 it can only be used with the CONNECT method.
Therefore, once both endpoints agree to use the Capsule Protocol, the frame
usage requirements of the stream change as specified in {{Section 8.5 of H2}}
and {{Section 4.2 of H3}}.
```

Author: Oh yes, that is much better
